app/lib/methods/getSettings.ts

```diff
@@ -146,16 +146,31 @@ export async function getSettings(): Promise<void> {
 		const db = database.active;
 		const settingsParams = Object.keys(defaultSettings).filter(key => !loginSettings.includes(key));
 		// RC 0.60.0
-		const result = await fetch(
-			`${sdk.current.client.host}/api/v1/settings.public?query={"_id":{"$in":${JSON.stringify(settingsParams)}}}&count=${
-				settingsParams.length
-			}`
-		).then(response => response.json());
-
-		if (!result.success) {
-			return;
-		}
-		const data: IData[] = result.settings || [];
+		let offset = 0;
+		let remaining;
+		let settings: IData[] = [];
+
+		// Iterate over paginated results to retrieve all settings
+		do {
+			// TODO: why is no-await-in-loop enforced in the first place?
+			/* eslint-disable no-await-in-loop */
+			const response = await fetch(
+				`${sdk.current.client.host}/api/v1/settings.public?query={"_id":{"$in":${JSON.stringify(settingsParams)}}}
+				&offset=${offset}`);
+
+			const result = await response.json();
+
+			if (!result.success) {
+				return;
```

Comment:
if you return in that case, without throwing an error, how can we know that the settings were not fetched? in this case, if one of the request fails, we should retry it until we get the result, and after a few retries, if it doesn't succeed, we should throw an error.

Reply:
> if you return in that case, without throwing an error, how can we know that the settings were not fetched? in this case, if one of the request fails, we should retry it until we get the result, and after a few retries, if it doesn't succeed, we should throw an error.

You are correct, but that was the behavior also prior to my PR. I want to focus on just one thing per PR.

Reply:
Can we resolve this conversation?

Reply:
Can you fix it? Although the issue was already there, you're refactoring the whole thing, so it's worth refactoring to a working state.

Reply:
Ok, no worries. We'll get there.
Gonna review your PR soon.
Thanks for your contribution 👊🏼